### ISSUE TITLE
How to change height, width, font size ... ?

### ISSUE THREAD

**Issue by klogx** [WhyNotHugo/python-barcode#17]
I only can change ImageWriter.dpi but font size not correct
![001](https://user-images.githubusercontent.com/12602709/47895673-42b19980-de9c-11e8-871f-cef752d88402.png)
(code128 & dpi = 100)
And set_options has no effect

**Comment by Sharlock93** [comment]
I had problems with setting options I wanted, pass the options parameter to the save function call not set_options

**Comment by WhyNotHugo** [comment]
Yeah, I think the API needs some simplification, having two ways of passing options is kinda pointless, and it's not the first time I've seen inconsistencies. I'll try to address it when I have a bit of time.

**Comment by Sharlock93** [comment]
Do you have any long term plans for this? any todo list?

**Comment by WhyNotHugo** [comment]
I don't have any defined plans yet. There's a few options I might consider:

* Making options actual kwarg.
* Removing options from either `set_options` or `save`. Not sure which one yet. It seems silly to have two ways of doing the same thing.

I've exams and other stuff going on right now, so won't be able to sit down and further address this for a while.

Finally, obviously I need to track down what's wrong that causing this specific bug and fixing that, but haven't looked at it yet.

**Comment by buzzlightmonth** [comment]
Since the `set_options` method doesn't work and isn't very Pythonic, I guess the `.save` should be the only officially supported method?

**Comment by lieryan** [comment]
I think we should fix `*Writer.set_options()` and remove `save(options)`.

The reason you'd be instantiating a custom `*Writer` is so that you can reuse a common `*Writer` options to generate multiple barcodes or to write a custom `*Writer` subclass. Passing options in the `*Writer` class is a simpler API that allows better extensibility and reusability than passing options in `save(options)`.

If you just want a quick one-liner to pass all the options in one go, then you probably should be using the `generate` shortcut method rather than `.get()+save()`. So IMO there's no real use cases to use `.get()+save()` that's not better served with `generate()`.

**Comment by WhyNotHugo** [comment]
My biggest concearn thus far has been to keep compatibility with `barcode`.

I think it's time to move past that, since it's proving impossible to fix some existing bugs with the existing API. too many duplicate functions, and too many ways to do the same thing.

I'm going to tag a new release with the current master, to enable addressing some of these issues (and accepting PRs). Otherwise this is just stagnating with unfixable issues in the name of backwards compatibility with a broken API.
